### PR TITLE
Raise an error for analyses which are not completed

### LINKED ISSUE #3297: The is_latest_analysis_completed check is ignored for storage
## Description

The `is_latest_analysis_completed` check is overlooked for storage. We expect it to throw an error if the analysis is incomplete, but it only returns a boolean value.

https://github.com/Clinical-Genomics/cg/blob/b796913674fa1ec6720b5be13f471f1cac06f191/cg/meta/workflow/nf_analysis.py#L779-L796

This affects all Nextflow workflows and Balsamic. However, it doesn't directly impact automation or production because we handle this check correctly when extracting cases for storage in `get_cases_to_store` within `store_available`.

## Suggested solution

Raise an exception after evaluating `is_latest_analysis_completed` during the storage step.

## This can be closed when

- The check is correctly used for storing cases.

### PR DESCRIPTION
**Pull request by islean**
## Description

Closes #3297.

### Changed

- An error is raised when the latest analysis has not completed in the nf-tower storing

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-store-validation -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/nf_analysis.py [hunk 1]
```diff
@@ -795,6 +800,10 @@ def store_analysis_housekeeper(self, case_id: str, dry_run: bool = False) -> Non
                 f"Could not store bundle in Housekeeper and StatusDB: {error}"
             )
 
+    def _verify_latest_analysis_is_completed(self, case_id: str) -> None:
+        if not self.trailblazer_api.is_latest_analysis_completed(case_id):
```
Comment (henrikstranneheim) [inline cg/meta/workflow/nf_analysis.py:804]: Add raise to doc str.

**Review by henrikstranneheim — APPROVED** [review]
Nice, 👍

File: cg/meta/workflow/nf_analysis.py [hunk 2]
```diff
@@ -795,6 +800,11 @@ def store_analysis_housekeeper(self, case_id: str, dry_run: bool = False) -> Non
                 f"Could not store bundle in Housekeeper and StatusDB: {error}"
             )
 
+    def _verify_latest_analysis_is_completed(self, case_id: str) -> None:
+        """Raises an AnalysisNotCompletedError if the latest analysis for the case has not completed"""
```
Comment (ivadym) [inline cg/meta/workflow/nf_analysis.py:804]:
```suggestion
        """Raises an AnalysisNotCompletedError if the latest analysis for the case has not completed."""
```

File: cg/meta/workflow/nf_analysis.py
Comment (ivadym) [inline cg/meta/workflow/nf_analysis.py:1]: Could you also fix it for the balsamic CLI? 🙏 

https://github.com/Clinical-Genomics/cg/blob/985acfbf7bf60718cefc3b54bf9ee3cc74d70662/cg/cli/workflow/balsamic/base.py#L139

**Review by ivadym — APPROVED** [review]
Nice! Thanks for looking into this! 💯

File: cg/meta/workflow/nf_analysis.py
Comment (islean) [reply cg/meta/workflow/nf_analysis.py:1]: Oh, I was not aware that this was a problem for other workflows as well. I'll have a look at them and see where the same change needs to be applied!

Comment (ivadym) [reply cg/meta/workflow/nf_analysis.py:1]: Yes, great! I've also found that it's used incorrectly in only those places. Thanks for the fix!

**Comment by islean** [comment]
Tested on stage:

```
[isak.ohlsson@hasta:workflow] [S_base] $ cg workflow tomte report-deliver <case_id>
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Case <case_id> exists in Status DB
Could not create report file: The latest analysis for <case_id> has not completed.
Aborted!
[isak.ohlsson@hasta:workflow] [S_base] $ 
```

**Comment by islean** [comment]
Deployed to production:

```
Your branch is behind 'origin/master' by 4 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating 2d6c3c9..e5c0c25
Fast-forward
 .bumpversion.cfg                         |  2 +-
 cg/__init__.py                           |  2 +-
 cg/apps/tb/api.py                        | 11 ++++++++++-
 cg/cli/workflow/balsamic/base.py         |  6 +++---
 cg/clients/freshdesk/__init__.py         |  0
 cg/clients/freshdesk/constants.py        | 21 +++++++++++++++++++++
 cg/clients/freshdesk/freshdesk_client.py | 38 ++++++++++++++++++++++++++++++++++++++
 cg/clients/freshdesk/models.py           | 36 ++++++++++++++++++++++++++++++++++++
 cg/exc.py                                |  6 ++++++
 cg/meta/workflow/nf_analysis.py          |  4 ++--
 pyproject.toml                           |  2 +-
 tests/mocks/tb_mock.py                   |  3 +++
 12 files changed, 122 insertions(+), 9 deletions(-)
 create mode 100644 cg/clients/freshdesk/__init__.py
 create mode 100644 cg/clients/freshdesk/constants.py
 create mode 100644 cg/clients/freshdesk/freshdesk_client.py
 create mode 100644 cg/clients/freshdesk/models.py
/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/cg/meta/workflow
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
[isak.ohlsson@hasta:workflow] [P_base] $ 
```

**Pull request merged**